src/ziplist.c

```diff
@@ -1477,6 +1477,8 @@ void ziplistRepr(unsigned char *zl) {
  * when `deep` is 1, we scan all the entries one by one. */
 int ziplistValidateIntegrity(unsigned char *zl, size_t size, int deep,
     ziplistValidateEntryCB entry_cb, void *cb_userdata) {
+    unsigned int header_count = intrev16ifbe(ZIPLIST_LENGTH(zl));
```

Comment:
same here, you must move this line a few lines below.

Reply:
Done.